sql/core/src/main/scala/org/apache/spark/sql/functions.scala

```diff
@@ -3381,6 +3381,48 @@ object functions {
     from_json(e, dataType, options)
   }
 
+  /**
+   * (Scala-specific) Parses a column containing a JSON string into a `MapType` with `StringType`
+   * as keys type, `StructType` or `ArrayType` of `StructType`s with the specified schema.
+   * Returns `null`, in the case of an unparseable string.
+   *
+   * @param e a string column containing JSON data.
+   * @param schema the schema to use when parsing the json string
+   *
+   * @group collection_funcs
+   * @since 2.4.0
+   */
+  def from_json(e: Column, schema: Column): Column = {
+    from_json(e, schema, Map.empty[String, String].asJava)
+  }
+
+  /**
+   * (Scala-specific) Parses a column containing a JSON string into a `MapType` with `StringType`
```

Comment:
Is it Scala specific or Java specific?

Reply:
I am not sure about this note at all. Why should it be Java or Scala specific?
I will change it to `Java-specific` to have it in the same style as other comments.

Reply:
Yup. That's because Java users will more likely use Java's collections rather than Scala's collection which works weirdly in Java side.

Reply:
Actually, can we remove this version for now and add it when it's requested? There is a concern about this file getting too long and `from_json` has so many variants.

Reply:
Maybe we will convert `object functions` to `package object functions` and put json related methods to a separate file. This `functions object` becomes really big.